Book-A-Desk.Domain/EventStore.fs

```diff
@@ -7,29 +7,54 @@ open Book_A_Desk.Domain.Reservation.Events
 
 type EventStore =
     {
-        GetEvents: Guid -> DomainEvent list
-        AppendEvents: Map<Guid, DomainEvent list> -> Result<unit, string>
+        GetEvents: Guid -> Result<DomainEvent list, string>
+        AppendEvents: Map<Guid, DomainEvent list> -> unit
     }
 
-module InMemoryEventStore =
-    open System.Collections.Generic
+module EventsReader =
+    let getEvents aggregateId events =
+        match Map.tryFind aggregateId events with
+        | Some events ->
+            Result.Ok events
+        | None ->
+            Result.Error "could not find aggregate id"
+  
+module EventsWriter =          
+    let appendEvents previousEventsById newEventsById =
+        newEventsById
+        |> Map.fold (fun (appendedEvents : Map<Guid, DomainEvent list>) aggregateId newEvents ->
+            appendedEvents.Add(aggregateId, newEvents)) previousEventsById
+
+module InMemoryEventStore =    
+    type private MailboxEventStore =
+    | GetEvents of Guid * AsyncReplyChannel<Result<DomainEvent list, string>>
+    | AppendEvents of Map<Guid, DomainEvent list>
+    
+    let private memoryStore =
+        MailboxProcessor.Start(fun inbox ->
```

Comment:
Simple explanation:
MailboxProcessors are essentially a queue that keeps a state and runs in their own thread. You can communicate with them by sending and receiving messages.

A side benefit of using this is concurrency. I did not make the memory store methods asynchronous but it is available.